datasets/austin_crime/_images/run_csv_transform_kub/csv_transform.py

```diff
@@ -0,0 +1,252 @@
+    df["longitude"] = (
+        df["address"]
+        .apply(search_string)
+        .apply(extract_lat_long, args=[r".*(\-\d+\.\d+)\)"])
+    )
+
+    logging.info("Transform: Creating a new column - year...")
+    df["location"] = "(" + df["latitude"] + "," + df["longitude"] + ")"
+
+    logging.info("Transform: Dropping column - address1...")
+    delete_column(df, "address1")
+
+    logging.info("Transform: Reordering headers...")
+    df = df[headers]
+
+    logging.info(f"Saving to output file.. {target_file}")
+    try:
+        save_to_new_file(df, file_path=str(target_file))
+    except Exception as e:
+        logging.error(f"Error saving output file: {e}.")
+
+    logging.info(
+        f"Uploading output file to.. gs://{target_gcs_bucket}/{target_gcs_path}"
+    )
+    upload_file_to_gcs(target_file, target_gcs_bucket, target_gcs_path)
+
+    logging.info(
+        "Austin crime process completed at "
+        + str(datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S"))
+    )
+
+
+def download_file(
+    source_url: typing.List[str], source_file: typing.List[pathlib.Path]
+) -> None:
+    for url, file in zip(source_url, source_file):
+        logging.info(f"Downloading file from {url} ...")
+        subprocess.check_call(["gsutil", "cp", f"{url}", f"{file}"])
+
+
+def read_files(path: pathlib.Path) -> pd.DataFrame:
+    all_files = glob.glob(path + "/*.csv")
+    li = []
+    for filename in all_files:
+        frame = pd.read_csv(filename, index_col=None, header=0)
+        li.append(frame)
+    df = pd.concat(li, axis=0, ignore_index=True)
+    return df
+
+
+def rename_headers(df: pd.DataFrame, rename_mappings: dict) -> None:
+    df.rename(columns=rename_mappings, inplace=True)
+
+
+def reg_exp_tranformation(str_value: str, search_pattern: str, replace_val: str) -> str:
+    str_value = re.sub(search_pattern, replace_val, str_value)
+    return str_value
+
+
+def convert_to_integer_string(input: typing.Union[str, float]) -> str:
+    str_val = ""
+    if not input or (math.isnan(input)):
+        str_val = ""
+    else:
+        str_val = str(int(round(input, 0)))
+    return str_val
+
+
+def convert_dt_format(dt_str: str) -> str:
+    a = ""
+    if dt_str is None or len(str(dt_str)) == 0 or str(dt_str) == "nan":
+        return str(a)
+    else:
+        return datetime.datetime.strptime(str(dt_str), "%m/%d/%Y %H:%M:%S %p").strftime(
+            "%Y-%m-%d %H:%M:%S"
+        )
+
+
+def extract_year(string_val: str) -> str:
+    string_val = string_val[0:4]
+    return string_val
+
+
+def convert_exp_to_float(exp_val: str) -> str:
+    float_val = "{:f}".format(float(exp_val))
+    return float_val
+
+
+def search_string(str_value: str) -> str:
+    a = ""
+    if re.search(r".*\(.*\)", str_value):
+        return str(str_value)
+    else:
+        return str(a)
+
+
+def extract_lat_long(str_val: str, patter: str) -> str:
+    m = re.match(patter, str_val)
+    if m:
+        return m.group(1)
```

Comment:
What happens if there's no match, what does the function return? Will that have any complications?

Reply:
added `else` condition